### DISCUSSION TITLE
Split up the code in ConsIndShockModel

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1665] [General]
There are a lot of different versions of AgentType in ConsIndShockModel and it can be hard to find one's place in the file by text search.

It would be helpful to break up PerfectForesight and ConsIndShock models into separate modules so it would be easier to know where one is.

**Comment by mnwhite** [comment]
Do we want to split these three models into separate files? PerfForesight, IndShock, and KinkedR? I prefer them in one file because  the models are substantially the same: they share a control, shock, and state space.
